pkg/manifests/manifests.go

```diff
@@ -2863,10 +2865,25 @@ func (f *Factory) ThanosRulerCustomResource(trustedCA *v1.ConfigMap) (*monv1.Tha
 		setEnv("NO_PROXY", f.config.HTTPConfig.NoProxy)
 	}
 
+	// Mounting TLS secret to thanos-ruler
+	if grpcTLS != nil {
+		secretName := "secret-grpc-tls"
+		secretVolume := v1.Volume{
+			Name: secretName,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: grpcTLS.GetName(),
+				},
+			},
+		}
+		t.Spec.Volumes = append(t.Spec.Volumes, secretVolume)
+	}
+	// TODO: error out if secret is nil
```

Comment:
not sure about this comment

Reply:
Yes, still need to implement this part, whenever the GRPC is nil we should error out otherwise thanos-ruler and thanos querier won't be able to establish communication.